### PR TITLE
Convert to Coffeescript

### PR DESCRIPTION
**Pull request by smizell**
Thoughts? Wrong move? Plain JavaScript wears me out now.

@fosrias @zdne

### CODE REVIEW

**Comment by zdne** [comment]
I am fine with it! Was just wondering what is the motivation..

**Comment by smizell** [comment]
1. Tired of writing JS
2. Curious
3. Want to extend some of these base types and prototypical inheritance is nasty

File: lib/primitives.js [hunk 1]
```diff
@@ -1,218 +1,258 @@
-var _ = require('lodash');
+var ArrayType, BoolType, ElementType, KeyValueType, NullType, NumberType, ObjectType, StringType, convertFromDom, convertToType, _,
```
Comment (zdne) [inline lib/primitives.js]: So I wonder – is this a standard procedure – bundling generated file with the package? Shouldn't this be generated during npm install?

Comment (smizell) [inline lib/primitives.js]: I have not followed this, but should before merging. 

http://stackoverflow.com/a/13646337/4175777

Comment (zdne) [inline lib/primitives.js]: See https://github.com/apiaryio/drafter.js/blob/master/package.json#L16 and https://github.com/apiaryio/drafter.js/blob/master/scripts/build

**Comment by smizell** [comment]
@zdne Fixed

File: gulpfile.coffee [hunk 2]
```diff
@@ -0,0 +1,23 @@
+jshint = require 'gulp-jshint'
+stylish = require 'jshint-stylish'
+mocha = require 'gulp-mocha'
+coffee = require 'gulp-coffee'
+gulp   = require 'gulp'
+
+gulp.task "lint", ->
```
Comment (zdne) [inline gulpfile.coffee:7]: Just out of curiosity: Why gulp instead of grunt?

**Comment by zdne** [comment]
@smizell alzo note I can't merge this since it is your repo :)

File: gulpfile.coffee (hunk 2 above)
Comment (smizell) [inline gulpfile.coffee:7]: Gulp works a ton better, as it is uses pipes to run tasks (which feels more like how Node stuff works), whereas Grunt used these crazy configuration objects.

**Pull request merged**